packages/core/src/types-hoist/span.ts

```diff
@@ -234,6 +234,19 @@ export interface Span {
 
   /**
    * Update the name of the span.
+   *
+   * **Important:** You most likely want to use `Sentry.updateSpanName(span, name)` instead!
+   *
+   * This method will update the current span name but cannot guarantee that the new name will be
+   * the final name of the span. Instrumentation might still overwrite the name with an automatically
+   * computed name, for example in `http.server` or `db` spans.
+   *
+   * You can ensure that your name is kept and not overwritten by calling `Sentry.updateSpanName(span, name)`
+   *
+   * If you want to update a span name in a browser-only app, `span.updateName` and `Sentry.updateSpanName`
+   * are identical: In both cases, the name will not be overwritten by the Sentry SDK.
```

Comment:
l: I would maybe omit this, as this is an implementation detail and may even change? But no strong feelings! 
```suggestion
```

Reply:
Yeah good call, let's omit this!

Reply:
[1772a77](https://github.com/getsentry/sentry-javascript/pull/14291/commits/1772a77a589b81d33bb3769ed1558fd6932654d1)